docs/cache/pattern/api-caching.md

```diff
@@ -0,0 +1,165 @@
+      return getResponse.valueString();
+    } else if (getResponse instanceof CacheGet.Miss) {
+      // If the weather is not in the Cache, call actual client,
+      // and populate the Cache.
+      const response = await this.delegate.getWeatherDescription(city);
+      await this.populateCacheWithResult(city, response);
+      return response;
+    } else if (getResponse instanceof CacheGet.Error) {
+      // If there is an error, simply fallback to calling the actual client.
+      return await this.delegate.getWeatherDescription(city);
+    } else {
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      throw new Error(`unrecognized getResponse ${getResponse}`);
+    }
+  }
+}
+```
+5. By using the Decorator Pattern, we can easily compose `CachingWeatherDescriptionClient` and `WolframWeatherDescriptionClient` together to create a new client that does read-side caching, i.e. if a result is the cache, return the result, otherwise fallback to retrieving the result from the remote API.
+This is an example code that leverages Momento Cache to cache results of WolframAlpha API into a remote cache:
+```typescript
+import {CacheClient, Configurations, CredentialProvider} from '@gomomento/sdk';
+
+async function main() {
+  const momentoClient = await CacheClient.create({
+    configuration: Configurations.Laptop.v1(),
+    credentialProvider: CredentialProvider.fromEnvironmentVariable({
+      environmentVariableName: 'MOMENTO_API_KEY',
+    }),
+    defaultTtlSeconds: 60,
+  });
+  const wolframWeatherClient = new WolframWeatherDescriptionClient();
+  const cachingWeatherClient = new CachingWeatherDescriptionClient(
+    wolframWeatherClient,
+    momentoClient
+  );
+
+  const response = await cachingWeatherClient.getWeatherDescription(
+    City.Phoenix
+  );
+
+  // As of this writing the weather in Arizona is:
+  // "The weather in Phoenix, Arizona, currently includes no precipitation with clear skies, a wind speed of 9 miles per hour and a temperature of 78 degrees Fahrenheit"
+  console.log(`${response}`);
+}
+
+```
+
+That is it! This is a simple example of how to cache an API result. Here's a few more things that you can tweak further to adapt Momento Cache to your exact use-case:
+1. You can further leverage the Decorator pattern to add more functionality without modifying the existing code. For example you can add a Decorator for metrics/logging so that you get instrumentation on error rates and cache hit rates.
+2. TTL (how long items get to live in a cache) can be adjusted on a per-item basis to fit your use-case. For example, in our earlier code, we could've made the TTLs different for different cities.
+3. You can cache more than just Strings into Momento Cache. In our earlier example we cache strings but Momento Cache can also accept byte arrays. This allows you to cache an arbitrary object in your application, provided that you implement your own custom serialization/deserialization.
```

Comment:
You should add a section down here similar to the *See More* section in the Topics token_id_webhook.md page for additional reading. I recommend these:

* title: 6 6ommon caching strategies
  * link: https://www.gomomento.com/blog/6-common-caching-design-patterns-to-execute-your-caching-strategy
  * description: Learn more about other types of caching strategies
*  title:  Using Momento to cache chatbot calls
  * link: https://www.gomomento.com/resources/case-studies/cydas-people-takes-momento-cache-to-prod-for-their-chatgpt-ai-powered-chatbot-in-just-two-hours
  * description: See how CYDAS used Momento to cache calls to LangChain to build a chatbot